components/unit/UnitVegaViz.vue

```diff
@@ -98,13 +106,13 @@ export default {
     }
   },
   watch: {
-    specWithItems(v) {
+    specWithItems() {
       this.draw()
     }
   },
   async mounted() {
-    this.width = this.$refs.graph.offsetWidth
-    await this.draw()
+    if (this.$refs.graph) this.width = this.$refs.graph.offsetWidth
+    if (this.isValid && !this.isEmpty) await this.draw()
```

Comment:
This is the part that seems the most hacky to me, what do you think?

Reply:
It looks ok to me. For the style, I would prefer to always use if with braces. I'm suprised that our over-pedantic linter doesn't enforce that. Also width can be 0, so I would write:
```

this.width = this.$refs.graph?.offsetWidth || 0
if (this.isValid && !this.isEmpty) {
   // we don't need to wait, actually
   this.draw() 
}
```